ui/appui-react/src/appui-react/framework/FrameworkDialogs.ts

```diff
@@ -0,0 +1,101 @@
+/*---------------------------------------------------------------------------------------------
+* Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+* See LICENSE.md in the project root for license terms and full copyright notice.
+*--------------------------------------------------------------------------------------------*/
+import { DialogChangedEvent, DialogInfo, DialogManagerBase } from "../dialog/DialogManagerBase";
+
+/** Modal Dialog Changed Event class.
+ * @public
+ */
+export class ModalDialogChangedEvent extends DialogChangedEvent { }
+
+/** Modeless Dialog Changed Event class.
+ * @public
+ */
+export class ModelessDialogChangedEvent extends DialogChangedEvent { }
+
+/** @public */
+export interface ModelessDialogInfo {
+  reactNode: React.ReactNode;
+  zIndex: number;
+  parentDocument: Document;
+}
+
+/**
+ * Manages dialog access
+ * @beta
+ */
+export interface FrameworkDialog {
+  /** @internal */
+  readonly dialogManager: DialogManagerBase;
```

Comment:
We should avoid internal properties in interfaces, because it makes them impossible to implement?

Reply:
I removed all `internal` and `deprecated` elements from the new interfaces, and the places which were calling those are now calling the implementation. We will want to make sure all of this is OK when we change implementation.